pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -205,12 +205,18 @@ func (cl *Client) Get(typ config.GroupVersionKind, name, namespace string) *conf
 }
 
 // Create implements store interface
-func (cl *Client) Create(config config.Config) (string, error) {
-	if config.Spec == nil {
-		return "", fmt.Errorf("nil spec for %v/%v", config.Name, config.Namespace)
+func (cl *Client) Create(cfg config.Config) (string, error) {
+	if cfg.Spec == nil {
+		return "", fmt.Errorf("nil spec for %v/%v", cfg.Name, cfg.Namespace)
+	}
+
+	// we probably dont want to error here because its nil in most
+	// use cases. just populate it with an empty status.
+	if cfg.Status == nil {
+		cfg.Status = new(config.Status)
```

Comment:
why can't we keep it nil?

Reply:
i thought when create() would later be called and a nil pointer dereference would happen:

```
			Status:     *(cfg.Status.(*v1alpha1.IstioStatus))
```